### PR TITLE
fix-17201

### LINKED ISSUE #17201: LICENSE: default-plugin uses API which is not licensed for commercial usage
### 📜 Description

The default-plugin generated by the CLI uses the API of https://randomuser.me: 

https://github.com/backstage/backstage/blob/master/packages/cli/templates/default-plugin/src/components/ExampleFetchComponent/ExampleFetchComponent.tsx.hbs#L79

- https://randomuser.me uses https://www.uifaces.co for generating data.
- https://www.uifaces.co/license in turn uses https://generated.photos/terms-and-conditions for generating photos
- https://generated.photos/terms-and-conditions is for non-commercial usage only (unless paid)

Why generating a plugin with CLI in an commercial context, this can be an issue.

### 👍 Expected behavior

The CLI should generate a plugin that is compatible with commercial environments.

### 👎 Actual Behavior with Screenshots

https://randomuser.me
image

https://www.uifaces.co/licenses/
image

https://generated.photos/terms-and-conditions
image

### 👟 Reproduction steps

Generate a new plugin with the CLI:
```shell
yarn new --select plugin
```

### 📃 Provide the context for the Bug.

Backstage Plugin development

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by npiyush97**
#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [X] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
fix #17201 
[Screencast from 06-06-23 08:05:28 AM IST.webm](https://github.com/backstage/backstage/assets/43876655/71541128-66e1-4a33-90b4-704b8bbaf99c)
Screenshot from 2023-06-08 15-00-13

### CODE REVIEW

File: .changeset/cyan-ants-pump.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'example-app': patch
```
Comment (jhaals) [inline .changeset/cyan-ants-pump.md:2]:
```suggestion
```

File: packages/cli/templates/default-plugin/src/components/ExampleFetchComponent/ExampleFetchComponent.tsx.hbs [hunk 2]
```diff
@@ -76,7 +66,7 @@ export const DenseTable = ({ users }: DenseTableProps) => {
 export const ExampleFetchComponent = () => {
   const { fetch } = useApi(fetchApiRef);
   const { value, loading, error } = useAsync(async (): Promise<User[]> => {
-    const response = await fetch('https://randomuser.me/api/?results=20');
+    const response = await fetch('https://tinyurl.com/fake-data');
```
Comment (jhaals) [inline packages/cli/templates/default-plugin/src/components/ExampleFetchComponent/ExampleFetchComponent.tsx.hbs:69]: Let's skip fetching of the contents entirely and place the data inside this example instead.

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR, added some tweaks :)

File: packages/cli/templates/default-plugin/src/components/ExampleFetchComponent/ExampleFetchComponent.tsx.hbs (hunk 2 above)
Comment (npiyush97) [reply packages/cli/templates/default-plugin/src/components/ExampleFetchComponent/ExampleFetchComponent.tsx.hbs:69]:
```
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Table, TableColumn, Progress, ResponseErrorPanel } from '@backstage/core-components';
import useAsync from 'react-use/lib/useAsync';
import Users from './fake-data.json'

const useStyles = makeStyles({
  avatar: {
    height: 32,
    width: 32,
    borderRadius: '50%',
  },
});

type User = {
  gender: string; // "male"
  name: {
    title: string; // "Mr",
    first: string; // "Duane",
    last: string; // "Reed"
  };
  email: string; // "duane.reed@example.com"
  picture: string; // "https://api.dicebear.com/6.x/open-peeps/svg?seed=Duane"
  nat: string; // "AU"
};

type DenseTableProps = {
  users: User[];
};

export const DenseTable = ({ users }: DenseTableProps) => {
  const classes = useStyles();

  const columns: TableColumn[] = [
    { title: 'Avatar', field: 'avatar' },
    { title: 'Name', field: 'name' },
    { title: 'Email', field: 'email' },
    { title: 'Nationality', field: 'nationality' },
  ];

  const data = users.map(user => {
    return {
      avatar: (
        <img
          src={user.picture}
          className={classes.avatar}
          alt={user.name.first}
        />
      ),
      name: `${user.name.first} ${user.name.last}`,
      email: user.email,
      nationality: user.nat,
    };
  });

  return (
    <Table
      title="Example User List"
      options={{ search: false, paging: false }}
      columns={columns}
      data={data}
    />
  );
};

export const ExampleFetchComponent = () => {

  const { value, loading, error } = useAsync(async (): Promise<User[]> => {
    const response = Users;
    const data = await response;
    return data.results;
  }, []);

  if (loading) {
    return <Progress />;
  } else if (error) {
    return <ResponseErrorPanel error={error} />;
  }

  return <DenseTable users={value || []} />;
};
```
I was thinking like this ?
@jhaals

File: .changeset/cyan-ants-pump.md (hunk 1 above)
Comment (npiyush97) [reply .changeset/cyan-ants-pump.md:2]: clarification question how does patch work in changeset?
I do yarn changeset select changed packages no on first major version select minor bump and summary ?
steps to do patch changeset?   
thank you

File: packages/cli/templates/default-plugin/src/components/ExampleFetchComponent/ExampleFetchComponent.tsx.hbs [hunk 3]
```diff
@@ -74,11 +288,11 @@ export const DenseTable = ({ users }: DenseTableProps) => {
...
 export const ExampleFetchComponent = () => {
-  const { fetch } = useApi(fetchApiRef);
+
   const { value, loading, error } = useAsync(async (): Promise<User[]> => {
-    const response = await fetch('https://randomuser.me/api/?results=20');
-    const data = await response.json();
-    return data.results;
+    const response = exampleUsers;
+    const data = response.results;
+    return data;
```
Comment (jhaals) [inline packages/cli/templates/default-plugin/src/components/ExampleFetchComponent/ExampleFetchComponent.tsx.hbs:295]:
```suggestion
    // Would use fetch in a real world example
    return exampleUsers.results;
```

Comment (jhaals) [reply packages/cli/templates/default-plugin/src/components/ExampleFetchComponent/ExampleFetchComponent.tsx.hbs:295]: Perhaps we can simplify this slightly and add a comment

**Review by jhaals — CHANGES_REQUESTED** [review]

File: packages/cli/templates/default-plugin/src/components/ExampleFetchComponent/ExampleFetchComponent.tsx.hbs (hunk 3 above)
Comment (npiyush97) [reply packages/cli/templates/default-plugin/src/components/ExampleFetchComponent/ExampleFetchComponent.tsx.hbs]:
```
const { value, loading, error } = useAsync(async (): Promise<User[]> => {
    // Simulating data fetching (e.g., using the 'fetch' function)
    return exampleUsers.results;
  }, []);
```
like this ?

**Comment by npiyush97** [comment]
@jhaals ready to merge ?

**Review by jhaals — APPROVED** [review]
Thank you

**Pull request merged**